client/js/components/tags/TagsListForm.vue

```diff
@@ -0,0 +1,262 @@
+        required />
+
+      <dp-select
+        v-model="newTag.topic"
+        class="mb-4"
+        data-cy="tagsList:newTag:topic"
+        :label="{
+          text: Translator.trans('entity.choose', { entity: Translator.trans('tag.category') })
+        }"
+        :options="topicsAsOptions"
+        required />
+
+      <addon-wrapper
+        class="block mb-4"
+        hook-name="tag.create.form"
+        @input="updateForm"
+        @change="updateForm" />
+
+      <dp-button-row
+        data-cy="tagsList:addNewTag"
+        primary
+        secondary
+        @primary-action="dpValidateAction('addNewTagForm', () => saveNewTag(), false)"
+        @secondary-action="closeForm" />
+    </div>
+
+    <div
+      v-else-if="currentForm === 'topic'"
+      id="new-topic-form"
+      class="border rounded p-4 my-4"
+      data-dp-validate="addNewTopicForm">
+      <dp-label
+        class="mb-4"
+        for="new-topic-form"
+        :text="Translator.trans('entity.create', { entity: Translator.trans('tag.category')})"
+        bold />
+      <dp-input
+        v-model="newTopic.title"
+        id="new-topic-title"
+        class="mb-4"
+        :label="{
+           text: Translator.trans('title')
+        }"
+        maxlength="250"
+        required />
+      <div class="flex justify-end mt-2">
+        <dp-button-row
+          data-cy="tagsList:addNewTopic"
+          primary
+          secondary
+          @primary-action="dpValidateAction('addNewTopicForm', () => saveNewTopic(), false)"
+          @secondary-action="closeForm" />
+      </div>
+    </div>
+
+    <div
+      v-else
+      class="flex gap-2 justify-end my-4">
+      <dp-button
+        data-cy="tagsList:openTagForm"
+        :text="Translator.trans('entity.create', { entity: Translator.trans('tag') })"
+        @click="() => openForm('tag')" />
+      <dp-button
+        data-cy="tagsList:openTopicForm"
+        :text="Translator.trans('entity.create', { entity: Translator.trans('tag.category') })"
+        variant="outline"
+        @click="() => openForm('topic')" />
+    </div>
+  </div>
+</template>
+
+<script>
+import { mapActions, mapMutations, mapState } from 'vuex'
+import AddonWrapper from '@DpJs/components/addon/AddonWrapper'
+import {
+  DpButton,
+  DpButtonRow,
+  DpInput,
+  DpLabel,
+  DpSelect,
+  dpValidateMixin
+} from '@demos-europe/demosplan-ui'
+
+export default {
+  name: 'TagsListForm',
+
+  components: {
+    AddonWrapper,
+    DpButton,
+    DpButtonRow,
+    DpInput,
+    DpLabel,
+    DpSelect,
+  },
+
+  mixins: [dpValidateMixin],
+
+  props: {
+    procedureId: {
+      type: String,
+      required: true
+    }
+  },
+
+  data () {
+    return {
+      currentForm: '',
+      newTag: {
+        title: '',
+        topic: ''
+      },
+      newTopic: {
+        title: ''
+      },
+      // This is necessary to allow extending the Tags-Resource
+      tagAttributes: {
+        boilerplate: '',
+        title: ''
+      }
```

Comment:
actually. This is not necessary, because wee don't extend the Tag-resource with addons. Addons hold their own reourceTypes which relate to the Tag-resourcetype in core. and boilerplate is an relationship anyway. I know, this is just copy-past from my stuff, but I think we should take the chance and clean it up.

Reply:
yes, i did not refactor this part, because i was not sure how you use it. done here https://github.com/demos-europe/demosplan-core/pull/4256/commits/a66f487c978895bde9aaa4336418b9f8781d59ec